gossip/src/crds_gossip_pull.rs

```diff
@@ -307,7 +304,9 @@ impl CrdsGossipPull {
                 let since = (now.saturating_sub(req_time).min(3600 * 1000) / 1024) as u32;
                 let stake = get_stake(&item.id, stakes);
                 let weight = get_weight(max_weight, since, stake);
-                (weight, item)
+                // Weights are bounded by max_weight defined above.
+                // So this type-cast should be safe.
+                ((weight * 100.0) as u64, item)
```

Comment:
Add a function for this calculation to use here and in crds_gossip_push.rs?

Reply:
Yeah, I think might be good to refactor `push_options` and `pull_options` to share the common parts. But I would rather not to mix that refactoring with this pull-request.